sa/storage-authority.go

```diff
@@ -193,6 +199,36 @@ func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationId int64
 	return
 }
 
+// GetCertificateByRequestID tries to look up a certificate based on
+// a random ID provided by the client.  It has the following output states:
+//
+// 1. No certificate record found with that identifier
+// 2. The identified certificate has been issued
+// 3. The identified certificate is still pending
+// 4. The identified certificate will never be issued
+//
+// The error return value will be non-nil only in the first case.  In the other
+// three cases, a certificate record will be returned, and the state of the certificate
+// will be indicated by the Certificate object returned.
+func (ssa *SQLStorageAuthority) GetCertificateByRequestID(requestID string) (cert core.Certificate, err error) {
+	// Request IDs are generated with core.NewToken, so they should look that way.
+	// As a side effect, this guards against rogue "%" characters.
+	if !core.LooksLikeAToken(requestID) {
+		err = errors.New("Invalid request ID " + requestID)
+		return
+	}
+
+	err = ssa.dbMap.SelectOne(&cert, "SELECT * FROM pending_cert WHERE requestID = :requestID",
```

Comment:
Why do `pending_cert` and `certificates` need to be two separate tables? Can't the certificate.Status column be used to indicate a pending cert and simplify these queries?